pkg/apis/autoscaling/validation/declarative_validation_test.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	genericapirequest "k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/registry/rest"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	featuregatetesting "k8s.io/component-base/featuregate/testing"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+	apitesting "k8s.io/kubernetes/pkg/api/testing"
+	"k8s.io/kubernetes/pkg/apis/autoscaling"
+	"k8s.io/kubernetes/pkg/features"
+)
+
+func TestValidateScaleForDeclarative(t *testing.T) {
+	ctx := genericapirequest.WithRequestInfo(genericapirequest.NewDefaultContext(), &genericapirequest.RequestInfo{
+		APIGroup:   "autoscaling",
+		APIVersion: "v1",
+	})
+	testCases := map[string]struct {
+		input        autoscaling.Scale
+		expectedErrs field.ErrorList
+	}{
+		// spec.replicas
+		"0 replicas": {
```

Comment:
In one of my PRs I switched to a convention of `fieldName: description` - less important here where there are few fields, but matters more when there are more fields and variations. 

Also, I think 0, positive, negative is enough - 1 feels redundant?

Reply:
I like it.  But I'll apply it here. I'll follow up on master, I don't see the convention there.